Doc/c-api/exceptions.rst

```diff
@@ -88,16 +88,29 @@ Printing and clearing
    The function is called with a single argument *obj* that identifies the context
    in which the unraisable exception occurred. If possible,
    the repr of *obj* will be printed in the warning message.
+   If *obj* is ``NULL``, only the traceback is printed.
 
    An exception must be set when calling this function.
 
+
+.. c:function:: void PyErr_FormatUnraisable(const char *format, ...)
+
+   Similar to :c:func:`PyErr_WriteUnraisable`, but the *format* and subsequent
+   parameters help format the warning message; they have the same meaning and
+   values as in :c:func:`PyUnicode_FromFormat`.
+   If *format* is ``NULL``, only the traceback is printed.
```

Comment:
Would you mind to give examples? I'm not sure how to use this API.

I understood that PyErr_WriteUnraisable(obj) can be written as:

```c
PyErr_FormatUnraisable("Exception ignored in: %R", obj);
```

And without an exception, it should be:

```c
PyErr_FormatUnraisable("Exception ignored in:");
```

Do the format must end with a newline character?

> If *format* is ``NULL``, only the traceback is printed.

What's the usage of this mode? Is the caller expected to log something like "Exception ignored in xxx" manually?

Reply:
`PyErr_FormatUnraisable(NULL)` is the same as `PyErr_WriteUnraisable(NULL)`. It just prints the traceback, without "Exception ignored ...".

I simply documented the existing behavior for `PyErr_WriteUnraisable()`, and for `PyErr_FormatUnraisable(NULL)` the most natural way is to do the same. What alternative do you propose?

Reply:
Sorry, my remark is not specific to NULL value, but more about the whole function. I don't know which string to pass to the function. I suggest to add an example to the doc. Like:

```
For example, ``PyErr_WriteUnraisable(obj)`` is similar to ``PyErr_FormatUnraisable("Exception ignored in: %R", obj);``.
```